### PR TITLE
Feature/rr 35 links in rich text

### PR DESCRIPTION
**Pull request by kkennedyAND**
## Description
Created a linkhandler component to identify wether a link is internal (and should use gatsby link) or an external link (which opens in a new tab) for use in rich text

## Changelog
Description about the modification

## Checks

- [ ] Responsive on mobile

- [ ] Implementation documented 

- [ ] 80% test coverage

- [ ] Ticket meets all Acceptance Criteria

- [ ] Matches/ aligns to content model

- [ ] Content model diagram updated

## Tested in 
- [ ] Chrome
- [ ] Safari

### CODE REVIEW

**Comment by Akallabet** [comment]
Some description for this PR?

File: gatsby-node.js [hunk 1]
```diff
@@ -71,19 +72,24 @@ exports.createPages = async ({graphql, actions}) => {
     })
   })
 
-  articles.forEach(({slug, section: {slug: sectionSlug}}) => {
-    const {article} = sections.find(({slug}) => slug === sectionSlug)
+  articles.forEach(({slug, section}) => {
```
Comment (Akallabet) [inline gatsby-node.js]: why did you change that?

File: src/components/LinkHandler/LinkHandler.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      <Link
+        to={url}
+        {...props}
+        activeClassName={activeclassname ? activeclassname : null}
+      >
+        {children}
+      </Link>
+    )
+  }
+  return newTab ? (
```
Comment (Akallabet) [inline src/components/LinkHandler/LinkHandler.js]: I'm not sure this is the right way to differentiate between external and internal link,
having this control at the top makes this component ambiguous, a component like that should be straight forward.
Also, with this component we have to refactor pretty much every other `Link` in other components.

I think the `if` should stay at the level where the two components can be used, which is RichText

File: src/components/RichText/FileLink.js [hunk 3]
```diff
@@ -36,11 +37,11 @@ const FileLink = ({title, url, contentType, size}) => {
   const fileSize = formatFileSize(size)
   return (
     <>
-      <a href={url} className={classes.link}>
+      <LinkHandler newTab url={url} className={classes.link}>
```
Comment (Akallabet) [inline src/components/RichText/FileLink.js]: This is where the control should happen

File: src/components/LinkHandler/LinkHandler.js (hunk 2 above)
Comment (andgreendigital) [reply src/components/LinkHandler/LinkHandler.js]: This follows the approach recommended by Gatsby in their docs: https://www.gatsbyjs.org/docs/gatsby-link/#reminder-use-link-only-for-internal-links
I like it because it abstracts away the complexity. The situation where Gatsby link is required is clearly defined and this component captures that definition so that the programmer doesn't have to consider it everywhere there is a link in the codebase.

Comment (Akallabet) [reply src/components/LinkHandler/LinkHandler.js]: I agree with the approach, I do not agree on where it is used

Comment (Akallabet) [reply src/components/LinkHandler/LinkHandler.js]: It should be confined within RichText

File: gatsby-node.js (hunk 1 above)
Comment (andgreendigital) [reply gatsby-node.js]: Because there was a bug where slugs were being given `/` in two places. We decided to add section slugs to page context in order to be consistent with article slugs.

Comment (ildikototh) [reply gatsby-node.js]: @andgreendigital we can properly setup  regexp to avoid any unnecessary  ```/```

File: src/components/LinkHandler/LinkHandler.js (hunk 2 above)
Comment (andgreendigital) [reply src/components/LinkHandler/LinkHandler.js]: What if I wanted to add a link to say... the footer... which went to an external source? Like a project partners website? With this approach that works straight out of the box. And it is also consistent throughout the codebase - whenever I need to use a link I use the LinkHandler it does the rest for me.

Comment (Akallabet) [reply src/components/LinkHandler/LinkHandler.js]: As of now, RichText is the only component where we have the internal/external link ambiguity.
If and when requirements change, we will adjust.
But for now we are refactoring pretty much each component only to take in account a specific requirement that comes only from RichText

Comment (andgreendigital) [reply src/components/LinkHandler/LinkHandler.js]: We are only swapping out Gatsby Link for a consistent interface which deals with this ambiguity. We could only use this component in the rich text but then we have 3 ways of handling links across the codebase instead of 1.

File: gatsby-node.js (hunk 1 above)
Comment (Akallabet) [reply gatsby-node.js]: As in my PR we can just remove the initial `/`, it's simpler

File: src/components/LinkHandler/LinkHandler.js (hunk 2 above)
Comment (Akallabet) [reply src/components/LinkHandler/LinkHandler.js]: ok let's do it

Comment (andgreendigital) [reply src/components/LinkHandler/LinkHandler.js]: Thanks Gratz! 😄

File: gatsby-node.js (hunk 1 above)
Comment (andgreendigital) [reply gatsby-node.js]: Okay - merge your PR first and we can change this one to match that convention. Also, could we consider enforcing that URL's start with a slash in Contentful?

Comment (andgreendigital) [reply gatsby-node.js]: Which part of that message are you thumbsing up?

Comment (Akallabet) [reply gatsby-node.js]: ah, the first part
